### PR TITLE
ruff: 0.0.265 -> 0.0.267

### PR DESCRIPTION
**Pull request by figsoda**
Diff: https://github.com/charliermarsh/ruff/compare/v0.0.265...v0.0.267

Changelog: https://github.com/charliermarsh/ruff/releases/tag/v0.0.267

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/ruff/default.nix [hunk 1]
```diff
@@ -1,33 +1,45 @@
...
   pname = "ruff";
-  version = "0.0.265";
+  version = "0.0.267";
 
   src = fetchFromGitHub {
     owner = "charliermarsh";
     repo = pname;
     rev = "v${version}";
-    hash = "sha256-b45kPvN2yILZBvP8eSNfWD6gpinwi9RsvmonRGYj0cI=";
+    hash = "sha256-0FMIjASw1DNJG0+4ERXTc6MKekOogSL5LsnEHZZNOYQ=";
```
Comment (mweinelt) [inline pkgs/development/tools/ruff/default.nix:18]:
```
error: hash mismatch in fixed-output derivation '/nix/store/i4agnywlf48spmgh9199f5nmqhsads3q-source.drv':
         specified: sha256-0FMIjASw1DNJG0+4ERXTc6MKekOogSL5LsnEHZZNOYQ=
            got:    sha256-inbW+oobW0hAsNdvJoiHvKoKAUjcuhEUrJe7fh5c6go=

Comment (figsoda) [reply pkgs/development/tools/ruff/default.nix]: not sure how that happened, there might have been a retag

this should be fixed now

**Comment by kalekseev** [comment]
This pr breaks `ruff-lsp`, it need to be updated to latest version too, please run `nixpkgs-review pr` before merging next time.

**Comment by mweinelt** [comment]
Turns out I did run nixpkgs-review, but since I'm on linux the tests for ruff-lsp weren't run, so nothing broke.

**Pull request merged**